packages/block-editor/src/utils/dom.js

```diff
@@ -27,7 +28,10 @@ export function isInsideRootBlock( blockElement, element ) {
 	const parentBlock = element.closest(
 		[ BLOCK_SELECTOR, APPENDER_SELECTOR ].join( ',' )
 	);
-	return parentBlock === blockElement;
+	return (
+		parentBlock === blockElement &&
+		! element.classList.contains( BLOCK_APPENDER_CLASS )
```

Comment:
Added this here to not focus block appender that is inside blocks such as group block. This helps us stay within the block itself.